src/clj/ataru/applications/excel_export.clj

```diff
@@ -551,10 +558,29 @@
         selected-hakukohde-oids      (or (some-> selected-hakukohde vector)
                                          (and selected-hakukohderyhma
                                               (hakukohderyhma-to-hakukohde-oids all-hakukohteet selected-hakukohderyhma)))
-        selected-hakukohderyhma-oids (or (some-> selected-hakukohderyhma vector)
-                                         (and selected-hakukohde
-                                              (hakukohde-to-hakukohderyhma-oids all-hakukohteet selected-hakukohde)))
-        selected-oids                (clojure.set/union (set selected-hakukohde-oids) (set selected-hakukohderyhma-oids))]
+        form-field-belongs-to        (let [belongs-not-specified? (fn [form-field]
+                                                                    (and (nil? (:belongs-to-hakukohderyhma form-field))
+                                                                         (nil? (:belongs-to-hakukohteet form-field))))]
+                                        (cond
+                                          (some? selected-hakukohde) (fn [form-field]
+                                                                        (or
+                                                                          (belongs-not-specified? form-field)
+                                                                          (contains?
+                                                                            (set (:belongs-to-hakukohteet form-field))
+                                                                            selected-hakukohde)
+                                                                          (let [hakukohderyhmas (hakukohde-to-hakukohderyhma-oids all-hakukohteet selected-hakukohde)]
+                                                                            (-> (clojure.set/intersection
+                                                                                  (set (:belongs-to-hakukohderyhma form-field))
+                                                                                  (set hakukohderyhmas))
+                                                                                empty?
+                                                                                not))))
+                                          (some? selected-hakukohderyhma) (fn [form-field]
+                                                                             (or
+                                                                               (belongs-not-specified? form-field)
+                                                                               (contains?
```

Comment:
Tässä pitää vielä tarkistaa sisältääkö kysymyksen `:belongs-to-hakukohteet` jonkin ryhmään `selected-hakukohderyhma` kuuluvan hakukohteen.

Reply:
Tehty